Sources/Testing/Events/Event.swift

```diff
@@ -123,16 +123,46 @@ public struct Event: Sendable {
   ///
   /// - Parameters:
   ///   - kind: The kind of event that occurred.
-  ///   - test: The test for which the event occurred, if any.
+  ///   - testID: The ID of the test for which the event occurred, if any.
   ///   - testCase: The test case for which the event occurred, if any.
   ///   - instant: The instant at which the event occurred. The default value
   ///     of this argument is `.now`.
-  init(_ kind: Kind, for test: Test? = .current, testCase: Test.Case? = .current, instant: Test.Clock.Instant = .now) {
+  ///
+  /// When creating an event to be posted, use
+  /// ``post(_:for:testCase:instant:configuration)`` instead since that ensures
+  /// any task local-derived values in the associated ``Event/Context`` match
+  /// the event.
+  init(_ kind: Kind, testID: Test.ID?, testCase: Test.Case?, instant: Test.Clock.Instant = .now) {
     self.kind = kind
-    self.test = test
+    self.testID = testID
```

Comment:
If the `Test` has moved to `Context`, we should consider moving the test case as well since it can carry an arbitrarily heavy payload in the form of arguments.

Reply:
I have now removed `Event.testCase` and left `Event.Context.testCase`. I plan to continue working in this area and find a way to replace the removed property with a lighter-weight value that indicates _which_ test case the event was for, without needing to include the full `Test.Case` value.